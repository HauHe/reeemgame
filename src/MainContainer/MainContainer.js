import React from "react";
export const MainContainer = () => <div>Hello world!</div>;
