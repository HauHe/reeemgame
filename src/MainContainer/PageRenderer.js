import React, { useEffect, useState } from 'react'
import ReactMarkdown from 'react-markdown'
import rehypeRaw from 'rehype-raw'
import rehypeSanitize from 'rehype-sanitize'
import styled from 'styled-components'

const Container = styled.div`
  padding: 0px 20px 20px 20px;
  display: flex;
  flex: 1;
  flex-wrap: wrap;
  align-content: flex-start;
  flex-direction: column;
`

export const Page = p => {
  const [content, setContent] = useState('')
  useEffect(() => {
    fetch(p.markdownFile)
      .then(response => response.text())
      .then(text => {
        setContent(text)
      })
  })
  return (
    <Container>
      <ReactMarkdown children={content} rehypePlugins={[rehypeRaw, rehypeSanitize]} />
    </Container>
  )
}
