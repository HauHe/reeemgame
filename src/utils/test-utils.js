import React from "react";
import PropTypes from "prop-types";
import { render } from "react-testing-library";
//import { ApolloProvider } from "react-apollo";
//import ContextStore from "../Context/ContextStore";
//import client from "../reeemApolloClient";
import { BrowserRouter } from "react-router-dom";

const AllTheProviders = ({ children }) => (
  // <ApolloProvider client={client}>
  //  <ContextStore>
  <BrowserRouter>{children}</BrowserRouter>
  //  </ContextStore>
  //</ApolloProvider>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "react-testing-library";

// override render method
export { customRender as render };

AllTheProviders.propTypes = {
  children: PropTypes.any
};
