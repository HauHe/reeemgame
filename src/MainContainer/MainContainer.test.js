import React from 'react'
import { render, cleanup } from 'test-utils'
import { MainContainer } from './MainContainer'
afterEach(cleanup)

it('should render correctly', () => {
  const { container } = render(<MainContainer />)
  expect(container).toMatchSnapshot()
})
