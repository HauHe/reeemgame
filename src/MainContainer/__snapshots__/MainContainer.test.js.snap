// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa bPGLpq"
  >
    <div
      class="sc-bwzfXH gkOoBg"
    >
      <div
        class="row no-gutters"
      >
        <div
          class="col"
        >
          <div
            class="sc-htpNat ktXyzA"
          >
            <a
              href="/"
            >
              <div
                class="sc-ifAKCX hrWLvt"
              >
                <img
                  alt="logo"
                  height="40"
                  src="REEEMlogo.transparent.1.svg"
                  width="200"
                />
              </div>
            </a>
            <div
              class="sc-bxivhb bAImMG"
            >
              game
            </div>
          </div>
        </div>
        <div
          class="col-lg-auto"
        >
          <div
            class="sc-bZQynM eHHkRo"
          >
            <a
              class="sc-gzVnrw iPHUbZ"
              href="/about"
            >
              About
            </a>
            <a
              class="sc-gzVnrw iPHUbZ"
              href="/score"
            >
              Score
            </a>
            <button
              class="sc-EHOje bbZGQD"
            >
              Share
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row no-gutters"
    >
      <div
        class="col-lg-auto"
      >
        <div
          class="sc-kgoBCf YejCR"
        >
          <div
            class="col"
          >
            <div
              class="row"
            >
              <div
                class="sc-kGXeez ljwDcV"
              >
                Electricity demands
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="sc-kGXeez ljwDcV"
              >
                Demand Profile
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="sc-kGXeez ljwDcV"
              >
                Technology Performance
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="sc-kGXeez ljwDcV"
              >
                Technology Cost
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="sc-kGXeez ljwDcV"
              >
                Generation constraints
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="sc-kGXeez ljwDcV"
              >
                Emission constraints
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="sc-kpOJdX bSlYGp"
              >
                <div
                  class="sc-cSHVUG isuxya"
                >
                  <img
                    alt="eu-flag"
                    class="sc-chPdSV bQdiiA"
                    src="/EU-flag.png"
                  />
                  <div
                    class="sc-kAzzGY dceIAl"
                  >
                    This project has received funding from the European Unionâ€™s Horizon 2020 research and innovation programme under grant agreement No 691739.
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="sc-VigVT fJOcCO"
        >
          <div
            class="col"
          >
            <h1
              class="sc-jTzLTM gjMEoS"
            >
              Goal
            </h1>
            <div
              class="sc-fjdhpX OqFYp"
            >
              Your goal is to maximise the score in 2050. Your score is calculated as the weighted average of the economic, social, and enviromental scores, which can be seen below.
            </div>
            <div
              class="container"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                />
                <div
                  class="col"
                >
                  Weight
                </div>
                <div
                  class="col"
                >
                  Score
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Economic
                </div>
                <div
                  class="col"
                >
                  21%
                </div>
                <div
                  class="col"
                >
                  OOOOOOOOOOOO
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Social
                </div>
                <div
                  class="col"
                >
                  38%
                </div>
                <div
                  class="col"
                >
                  OOOOO
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Envionmental
                </div>
                <div
                  class="col"
                >
                  41%
                </div>
                <div
                  class="col"
                >
                  OOOOOOOO
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-iwsKbI istRUX"
        >
          <div
            class="col"
          >
            <h1
              class="sc-gZMcBi pUXdS"
            >
              Decisions in 2020
            </h1>
            <div
              class="sc-gqjmRU fAMbXr"
            >
              Which decisions do you want to make in order to reach your goal?
            </div>
            <form
              class=""
            >
              <div
                class="form-group"
              >
                <div>
                  Spending on energy technology (R
                  &
                  D)?
                </div>
                <div
                  class="form-check"
                >
                  <input
                    class="form-check-input"
                    id="high"
                    name="research"
                    type="radio"
                  />
                  <label
                    class="form-check-label"
                    for="high"
                    title=""
                    type="checkbox"
                  >
                    High
                  </label>
                </div>
                <div
                  class="form-check"
                >
                  <input
                    class="form-check-input"
                    id="low"
                    name="research"
                    type="radio"
                  />
                  <label
                    class="form-check-label"
                    for="low"
                    title=""
                    type="checkbox"
                  >
                    Low
                  </label>
                </div>
              </div>
              <div
                class="form-group"
              >
                <div>
                  Emission restrictions?
                </div>
                <div
                  class="form-check"
                >
                  <input
                    class="form-check-input"
                    id="high"
                    name="emissions"
                    type="radio"
                  />
                  <label
                    class="form-check-label"
                    for="high"
                    title=""
                    type="checkbox"
                  >
                    High
                  </label>
                </div>
                <div
                  class="form-check"
                >
                  <input
                    class="form-check-input"
                    id="med"
                    name="emissions"
                    type="radio"
                  />
                  <label
                    class="form-check-label"
                    for="med"
                    title=""
                    type="checkbox"
                  >
                    Medium
                  </label>
                </div>
                <div
                  class="form-check"
                >
                  <input
                    class="form-check-input"
                    id="low"
                    name="emissions"
                    type="radio"
                  />
                  <label
                    class="form-check-label"
                    for="low"
                    title=""
                    type="checkbox"
                  >
                    Low
                  </label>
                </div>
              </div>
              <button
                class="btn btn-outline-success"
                type="submit"
              >
                Submit decisions
              </button>
            </form>
          </div>
        </div>
      </div>
      <div
        class="col-lg-auto"
      >
        <div
          class="sc-jzJRlG dPOBZH"
        >
          <div
            class="row no-gutters"
          >
            <img
              alt="Europe"
              src="europe.svg"
            />
          </div>
        </div>
        <div
          class="sc-dxgOiQ fhkfld"
        >
          <div
            class="progress"
          >
            <div
              aria-valuemax="2050"
              aria-valuemin="2015"
              aria-valuenow="2019"
              class="progress-bar"
              role="progressbar"
              style="width: 11.429%;"
            >
              2019
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
